### ISSUE TITLE
UnnecessaryFullyQualifiedName false positive with two classes with the same name in different packages

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#9322]
## Expected Behavior
No issue raised

## Observed Behavior
Detekt raise an issue

## Steps to Reproduce
Here there are 2 failing unit tests:
```kotlin
        @Test
        fun `does not report annotation call when shadowed`() {
            val code = """
                package foo

                import foo.bar1.Foo

                @foo.bar2.Foo
                object Bar : Foo
            """.trimIndent()
            val annotationCode = """
                package foo.bar1
                interface Foo
            """.trimIndent()
            val interfaceCode = """
                package foo.bar2
                annotation class Foo
            """.trimIndent()

            assertThat(subject.lintWithContext(env, code, annotationCode, interfaceCode)).isEmpty()
        }

        @Test
        fun `does not report annotation call when shadowed 2`() {
            val code = """
                package foo

                import foo.bar2.Foo

                @Foo
                object Bar : foo.bar1.Foo
            """.trimIndent()
            val annotationCode = """
                package foo.bar1
                interface Foo
            """.trimIndent()
            val interfaceCode = """
                package foo.bar2
                annotation class Foo
            """.trimIndent()

            assertThat(subject.lintWithContext(env, code, annotationCode, interfaceCode)).isEmpty()
        }
```

## Your Environment
* Version of detekt used: 2.2.0.0-alpha.3 (and current main)
